src/Quadrature/quadrature.jl

```diff
@@ -42,18 +42,19 @@ julia> getpoints(qr)
  [0.33333333333333, 0.33333333333333]
 ```
 """
-struct QuadratureRule{shape,T,dim}
-    weights::Vector{T}
-    points::Vector{Vec{dim,T}}
-    function QuadratureRule{shape, T}(weights::Vector{T}, points::Vector{Vec{dim, T}}) where {dim, shape <: AbstractRefShape{dim}, T}
+# struct QuadratureRule{shape, T, rdim, WeightStorageType <: AbstractVector{T}, PointStorageType <: AbstractVector{<:Vec{rdim,T}}}
+struct QuadratureRule{shape, T, rdim, WeightStorageType <: AbstractVector{T}, PointStorageType <: AbstractVector} # The one above crashes the deprecation in deprecations.jl for some reason...
```

Comment:
```julia
struct QuadratureRule{RefShape, WT, PT}
    weights::WT
    points::PT
    function QuadratureRule{RefShape}(weights::AbstractVector{T}, points::AbstractVector{Vec{rdim, T}}) where {rdim, RefShape <: AbstractRefShape{rdim}, T}
        if length(weights) != length(points)
            throw(ArgumentError("number of weights and number of points do not match"))
        end
        new{RefShape, typeof(weights), typeof(points)}(weights, points)
    end
end
```

Reply:
And change all (L60-L155)
```diff
- function QuadratureRule{ConcreteRefShape, T}(quad_type::Symbol, order::Int) where T
+ function QuadratureRule{ConcreteRefShape}(::Type{T}, quad_type::Symbol, order::Int) where {T<:Number}
```

Reply:
So if I understand your suggestion correctly, then you would remove all type constraints in the type parametrization?

Reply:
Yes, but these should be checked in the inner constructor.

Reply:
(I left one in `FacetQuadratureRule` but that was just since it makes it more readable IMO, but that could also just be checked in the inner constructor of course)